conda-store-server/conda_store_server/worker/tasks.py

```diff
@@ -43,6 +44,17 @@ def worker(self):
 
             self._worker.initialize()
 
+        # Installs a signal handler that terminates the process on Ctrl-C
+        # (SIGINT)
+        # Note: the following would not be enough to terminate beat and other
+        # tasks, which is why the code below explicitly calls sys.exit
+        # > from celery.apps.worker import install_worker_term_hard_handler
+        # > install_worker_term_hard_handler(self._worker, sig='SIGINT')
+        def _shutdown(*args, **kwargs):
+            return sys.exit(1)
+
+        platforms.signals["INT"] = _shutdown
```

Comment:
is sigint the only signal we need to worry about here? Do we face similar issues with sigterm or sigkill? Do those need handled too?

Reply:
I don't think we need to touch those. `SIGINT` is special because it can be sent via Ctrl-C. The two signals you mention need to be sent via something like `kill` or `pkill`, at which point the user needs to provide a proper process name or process id. Also, Celery has [pre-defined handlers for SIGTERM and SIGKILL](https://docs.celeryq.dev/en/stable/userguide/workers.html#stopping-the-worker), which correspond to soft and cold shutdowns, so I don't want to override that behavior.